src/Parser/Mage/Fire/Modules/Features/HotStreak.js

```diff
@@ -30,115 +32,182 @@ class HotStreak extends Analyzer {
     const spellId = event.ability.guid;
     if (spellId !== SPELLS.HOT_STREAK.id) {
       return;
     }
+    //If Hot Streak is removed and re-applied within 100ms of eachother then Pyromaniac Proc'd and granted a new hot streak
+    if (this.combatants.selected.hasTalent(SPELLS.PYROMANIAC_TALENT.id) && this.buffAppliedTimestamp - this.hotStreakRemoved < PROC_WINDOW_MS) {
+      this.pyromaniacProc = true;
+    }
     this.totalProcs += 1;
   }
 
-  on_byPlayer_removebuff(event) {
+  on_toPlayer_removebuff(event) {
     const spellId = event.ability.guid;
-    if (spellId !== SPELLS.HOT_STREAK.id) {
+    if (spellId !== SPELLS.HOT_STREAK.id && spellId !== SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
       return;
     }
+    if (spellId === SPELLS.HOT_STREAK.id) {
+      this.hotStreakRemoved = event.timestamp;
+    } else if (spellId === SPELLS.KAELTHAS_ULTIMATE_ABILITY.id) {
+      this.bracerProcRemoved = event.timestamp;
+      return;
+    }
+
+    //Check for Expired Procs, also if Fireball or Scorch was cast at the same time that Hot Streak was removed, then it was cast alongside Hot Streak (For the cast before hot streak check). Excluded Hot Streak procs during Combustion.
+    //Also checks to see if the bracer proc was removed within 100ms of Hot Streak getting removed (i.e. they hard casted Pyroblast for the bracer proc and used Hot Streak on the end of it.)
     if (!this.lastCastTimestamp || this.lastCastTimestamp + PROC_WINDOW_MS < this.owner.currentTimestamp) {
       this.expiredProcs += 1;
-      if (debug) {
-        console.log("Hot Streak proc expired @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
-      }
+      debug && console.log("Hot Streak proc expired @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
+    } else if (this.hotStreakRemoved - PROC_WINDOW_MS < this.castTimestamp || this.hotStreakRemoved - PROC_WINDOW_MS < this.bracerProcRemoved) {
+      this.castedBeforeHotStreak += 1;
+    } else if (!this.combatants.selected.hasBuff(SPELLS.COMBUSTION.id)) {
+      this.noCastBeforeHotStreak += 1;
+      debug && console.log("No hard cast before Hot Streak @ " + formatMilliseconds(event.timestamp - this.owner.fight.start_time));
     }
   }
 
   get usedProcs() {
     return this.totalProcs - this.expiredProcs;
   }
 
+  get expiredProcsPercent() {
+    return (this.expiredProcs / this.totalProcs) || 0;
+  }
+
+  get hotStreakUtil() {
+    return 1 - (this.expiredProcs / this.totalProcs) || 0;
+  }
+
+  get castBeforeHotStreakUtil() {
+    return 1 - (this.noCastBeforeHotStreak / (this.castedBeforeHotStreak + this.noCastBeforeHotStreak));
+  }
+
+  get expiredProcsThresholds() {
+    return {
+      actual: this.expiredProcsPercent,
+      isGreaterThan: {
+        minor: 0.05,
+        average: 0.10,
+        major: 0.20,
+      },
+      style: 'percentage',
+    };
+  }
+
+  get wastedCritsThresholds() {
+    return {
+      actual: this.wastedCrits,
```

Comment:
Don't make suggestions based on absolute number... this looks like an appropriate time to use wasted crits per minute.